.github/workflows/build-and-release-runners.yml

```
@@ -66,6 +72,6 @@ jobs:
             RUNNER_VERSION=${{ env.RUNNER_VERSION }}
             DOCKER_VERSION=${{ env.DOCKER_VERSION }}
           tags: |
-            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:v${{ env.RUNNER_VERSION }}
-            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:v${{ env.RUNNER_VERSION }}-${{ steps.vars.outputs.sha_short }}
-            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:latest
+            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:ubuntu-${{ matrix.os-version }}-v${{ env.RUNNER_VERSION }}
+            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:ubuntu-${{ matrix.os-version }}-v${{ env.RUNNER_VERSION }}-${{ steps.vars.outputs.sha_short }}
+            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:ubuntu-latest
```

Thanks for the input!!

I took a few days to this about what tagging strategy work best for us. Following conventions seen in popular docker images like https://hub.docker.com/_/docker and https://hub.docker.com/_/golang, I got to think we'd better use...:

```suggestion
            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:v${{ env.RUNNER_VERSION }}-ubuntu-${{ matrix.os-version }}
            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:v${{ env.RUNNER_VERSION }}-${{ steps.vars.outputs.sha_short }}-ubuntu-${{ matrix.os-version }}
            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:ubuntu-${{ matrix.os-version }}
```

Also, I think including `${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:ubuntu-latest`, a.k.a` laetst` here could result in the `ubuntu-latest` tag unstable(can randomly change between be 18.04 or 20.04, where we would expect it to consistently be 20.04), due to that it runs on both `os-version: 20.04` and `os-version: 18.04` for `name: actions-runner`.

---

I haven't tested it yet myself but probably we need a dedicated step for building `ubuntu-latest` a.k.a` `latest`, like below? Note that I added `if: ${{ matrix.os-version == '20.04' }}` so that it only runs once per workflow run.

```
      - name: Build and Push (latest)
        uses: docker/build-push-action@v2
        with:
          context: ./runner
          file: ./runner/${{ matrix.dockerfile }}
          platforms: linux/amd64,linux/arm64
          push: ${{ github.event_name != 'pull_request' }}
          if: ${{ matrix.os-version == '20.04' }}
          build-args: |
            RUNNER_VERSION=${{ env.RUNNER_VERSION }}
            DOCKER_VERSION=${{ env.DOCKER_VERSION }}
          tags: |
            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:v${{ env.RUNNER_VERSION }}
            ${{ env.DOCKERHUB_USERNAME }}/${{ matrix.name }}:latest
```

---

Alright! So in conclusion tagging wise we are going with this format:

`actions-runner:v2.277.1-ubuntu-18.04`
`actions-runner:v2.277.1-ubuntu-20.04`
`actions-runner:latest` <--- this is going to be considered ubuntu regardless of if more OS types get introduced. Ubuntu is considered our default runner and so latest is implicitly OS type `ubuntu`. This also ensures we keep backwards compatibility too and no weirdness with docker standards.

I think you are saying we won't bother with an `ubuntu-latest` right, keep regular `latest` instead for the above reasons?

---

Exactly! Thanks for summarizing